### DISCUSSION TITLE
Not understanding Item Progression mode

### DISCUSSION THREAD

**Discussion by onlyheretotest** [Malcolm-Q/LC-LateGameUpgrades#550] [Q&A] [answered]
From what I'm understanding, CustomScrap Item Progression makes it so that when you sell any of the listed scrap items to the company, a % of their total cost is put towards progressing the upgrade. However, when I sell any of the scrap items I put down there, it doesn't seem that any progress is made, nor does the cost of the upgrade change at all. Am I misunderstanding something or doing something wrong? Thanks in advance!

**Comment by WhiteSpike** [comment, marked as answer]
CustomScrap uses the names inserted on each upgrades configuration to tell which contribute and which do not.
Right now, it needs to be lower case to properly work and the names used are either the names displayed in their scan nodes or their internal names in the game code.
Once sold and found the association, it will reduce the price of the upgrade til it reaches zero and rank up.
Unique Scrap is the variant of this where it assigns all items to random upgrades.

**Reply by onlyheretotest** [reply]
Thank you! Two more questions, since I'm utilizing more than one item per upgrade, will it work if I format it like:
ring,perfume bottle,painting,gold bar,golden cup,fancy lamp,cash register
Will this work or does it have to be a different way still?
Last one, do special characters (such as the - in V-type engine and the space present in 2+ letter scraps) stop the item from registering as well?

**Reply by WhiteSpike** [reply]
That should work for multiple as it separates each item between commas.

It should still include the characters you inserted unless C#/Unity decides not to.
